File: Strong.py
import math
def is_strong_number(num):
    sum_of_factorials = 0
    temp = num
    while temp > 0:
        digit = temp % 10
        sum_of_factorials += math.factorial(digit)
        temp //= 10
    return sum_of_factorials == num

# Print strong numbers from 1 to 5000
print("Strong numbers from 1 to 5000 are:")
for i in range(1, 5001):
    if is_strong_number(i):
        print(i)

File: strong.java
public class Strong{
    // Precompute factorials for digits 0–9
    private static final int[] FACTORIAL = new int[10];
    static {
        FACTORIAL[0] = 1;
        for (int i = 1; i < 10; i++) {
            FACTORIAL[i] = FACTORIAL[i - 1] * i;
        }
    }

    // Check if a number is strong
    public static boolean isStrong(int num) {
        int original = num;
        int sum = 0;
        while (num > 0) {
            int digit = num % 10;
            sum += FACTORIAL[digit];
            num /= 10;
        }
        return sum == original;
    }

    public static void main(String[] args) {
        System.out.println("Strong numbers from 1 to 5000:");
        for (int i = 1; i <= 5000; i++) {
            if (isStrong(i)) {
                System.out.println(i);
            }
        }
    }
}